Unclear stacktrace on server startup

**Describe the bug**
After update my project from 0.23.2 to 0.25.0 and all keycloak configurations, the server did not started. My keycloak was up, however without my realm.

**Expected behavior**
StackTrace with a clear message, like `please make sure the 'quarkus.oidc.auth-server-url' is correct`

**Actual behavior**
(Describe the actual behavior clearly and concisely.)

```2019-10-16 19:06:46,165 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 1912ms
java.util.concurrent.CompletionException: io.vertx.core.impl.NoStackTraceThrowable: Not Found
        at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412)
        at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2044)
        at io.quarkus.oidc.VertxKeycloakRecorder.setup(VertxKeycloakRecorder.java:59)
        at io.quarkus.deployment.steps.VertxKeycloakBuildStep$setup44.deploy_0(VertxKeycloakBuildStep$setup44.zig:73)
        at io.quarkus.deployment.steps.VertxKeycloakBuildStep$setup44.deploy(VertxKeycloakBuildStep$setup44.zig:92)
        at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:169)
        at io.quarkus.runtime.Application.start(Application.java:94)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:135)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:180)
        at io.quarkus.dev.DevModeMain.start(DevModeMain.java:94)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:66)
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Not Found
2019-10-16 19:06:48,149 ERROR [io.qua.dev.DevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:225)
        at io.quarkus.runtime.Application.start(Application.java:94)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:135)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:180)
        at io.quarkus.dev.DevModeMain.start(DevModeMain.java:94)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:66)
Caused by: java.util.concurrent.CompletionException: io.vertx.core.impl.NoStackTraceThrowable: Not Found
        at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412)
        at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2044)
        at io.quarkus.oidc.VertxKeycloakRecorder.setup(VertxKeycloakRecorder.java:59)
        at io.quarkus.deployment.steps.VertxKeycloakBuildStep$setup44.deploy_0(VertxKeycloakBuildStep$setup44.zig:73)
        at io.quarkus.deployment.steps.VertxKeycloakBuildStep$setup44.deploy(VertxKeycloakBuildStep$setup44.zig:92)
        at io.quarkus.runner.ApplicationImpl1.doStart(ApplicationImpl1.zig:169)
        ... 5 more
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Not Found

2019-10-```

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OIDCException.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OIDCException.java
@@ -1,0 +1,19 @@
+package io.quarkus.oidc;
+
+@SuppressWarnings("serial")
+public class OIDCException extends RuntimeException {
+    public OIDCException() {
+    }
+
+    public OIDCException(String message) {
+        this(message, null);
+    }
+
+    public OIDCException(Throwable cause) {
+        this(null, cause);
+    }
+
+    public OIDCException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}

--- a/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/VertxKeycloakRecorder.java
+++ b/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/VertxKeycloakRecorder.java
@@ -49,7 +49,7 @@ public class VertxKeycloakRecorder {
             @Override
             public void handle(AsyncResult<OAuth2Auth> event) {
                 if (event.failed()) {
-                    cf.completeExceptionally(event.cause());
+                    cf.completeExceptionally(toOidcException(event.cause()));
                 } else {
                     cf.complete(event.result());
                 }
@@ -62,4 +62,12 @@ public class VertxKeycloakRecorder {
         mechanism.setAuth(auth);
         mechanism.setAuthServerURI(config.authServerUrl);
     }
+
+    protected static OIDCException toOidcException(Throwable cause) {
+        final String message = "OIDC server is not available at the 'quarkus.oidc.auth-server-url' URL. "
+                + "Please make sure it is correct. Note it has to end with a realm value if you work with Keycloak, for example:"
+                + " 'https://localhost:8180/auth/realms/quarkus'";
+        return new OIDCException(message, cause);
+    }
+
 }

